### DISCUSSION TITLE
Setting webproject to https locally, blazor websocket not upgrading

### DISCUSSION THREAD

**Discussion by mip1983** [microsoft/aspire#2266] [Q&A]
Hi there,

In my aspire app I set my Blazor Web project to have an https endpoint (for debugging against my oidc server) i.e.

```
builder.AddProject<Projects.yourprojecthere_Web>("web")
	.WithHttpsEndpoint(hostPort: 7167)

```

This seems to generally work fine, except with Blazor server components where it's establishing a websocket connection, the response is 200 rather than 101. While it operates, I get a few signalR dropouts this way and would rather it got the full fat websocket connection, which seems to work better if I switch back to regular http.

![image](https://github.com/dotnet/aspire/assets/2116108/fcc1bfa6-f7c1-4bab-b7f3-4b518dfbbb6f)

Is there any additional config I can add to allow websockets to work with https locally?

If it helps, this is the more detailed error I get with 'just my code' disabled:

> System.Net.WebSockets.WebSocketException: 'The remote party closed the WebSocket connection without completing the close handshake.'
> 
> Win32Exception: The specified data could not be decrypted.
> IOException: The decryption operation failed, see inner exception.
> ConnectionAbortedException: The HTTP/2 connection faulted.
> IOException: The request stream was aborted.

**Comment by davidfowl** [comment]
This should just work. Does it reproduce consistently?

**Reply by mip1983** [reply]
Yea, consistently.

If I file > new an Aspire Starter template project in VS2022 Preview and either update the launch profile or put '.WithHttpsEndpoint();' on the project, run it and go to the counter link, I see this on the https endpoint version:

![image](https://github.com/dotnet/aspire/assets/2116108/4432fd79-b864-4f61-9141-b99745b4b2c1)

On the http endpoint I get the 101 connection upgraded:

![image](https://github.com/dotnet/aspire/assets/2116108/5011d23c-78a0-401e-9950-1865f47d3779)

**Comment by iamshen** [comment]
i got the same problem

![image](https://github.com/dotnet/aspire/assets/15643938/8b028f72-0500-486a-bdcb-f53d7dd1be92)


![image](https://github.com/dotnet/aspire/assets/15643938/93f5047a-f6d7-4734-a042-ff5b6b274347)


![image](https://github.com/dotnet/aspire/assets/15643938/2611789d-08e5-4b5c-b07a-6d6655743d06)
